src/oauth-provider.ts

```diff
@@ -2068,6 +2140,21 @@ class OAuthProviderImpl {
     return new OAuthHelpersImpl(env, this);
   }
 
+  /**
+   * Saves a grant to KV with appropriate TTL based on expiration
+   * @param env - The environment bindings
+   * @param grantKey - The KV key for the grant
+   * @param grantData - The grant data to save
+   * @param now - Current timestamp in seconds
+   */
+  private async saveGrantWithTTL(env: any, grantKey: string, grantData: Grant, now: number): Promise<void> {
+    // Calculate TTL: undefined if no expiration or already expired (shouldn't happen but handled gracefully)
+    const grantTTL = grantData.expiresAt && grantData.expiresAt > now ? grantData.expiresAt - now : undefined;
```

Comment:
I think if `grantData.expiresAt` is before `now` -- i.e. the grant is already expired -- we probably want to delete the record, rather than save it with no TTL?